PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```diff
@@ -0,0 +1,135 @@
+        if (bukkitPlayer == null) return;
+        Optional<Inventory> oTeamInventory = getTeamsInventory(bukkitPlayer);
+
+        oTeamInventory.ifPresent(teamInventory -> {
+            event.setCancelled(true);
+            // If the item is in the off-hand slot, it wont get put back visually for the player without this.
+            if(event.getHand() == EquipmentSlot.OFF_HAND) event.getActor().updateInventory();
+            bukkitPlayer.openInventory(teamInventory);
+
+        });
+    }
+
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onInventoryClick(InventoryClickEvent event) {
+        if (event.getCurrentItem() == null) return;
+
+        // No putting evil items (ender chest, possibly wool) into the chest
+        Optional<Inventory> teamChest = getTeamsInventory(event.getActor());
+        if (teamChest.filter(teamInventory -> {
+            return teamInventory.equals(event.getView().getTopInventory());
+        }).isPresent() &&
+                isItemEvil(event.getCurrentItem())) {
+            event.setCancelled(true);
+            return;
+        }
+
+        // If normal right click, in their inventory, on the chest, then open shared inventory.
+        getTeamsInventory(event.getActor()).ifPresent(teamInventory -> {
+            if (event.getInventory().getType() == InventoryType.CRAFTING &&
+                    event.getCurrentItem().getType() == TOOL_TYPE &&
+                    event.getAction() == InventoryAction.PICKUP_HALF) {
+                event.setCancelled(true);
+                // This resets their mouse position annoyingly, but without it items can get stuck in places.
+                event.getActor().closeInventory();
+                // Prevent visual inconsistencies, specifically off-hand slot
+                if (event.getSlotType() == InventoryType.SlotType.QUICKBAR) {
+                    event.getActor().updateInventory();
+                }
+                event.getActor().openInventory(teamInventory);
+            }
+        });
+    }
+
+    private boolean isItemEvil(ItemStack item) {
+        if(item.getType() == TOOL_TYPE) return true;
+        if(oWmm.filter(wmm -> wmm.isObjectiveWool(item)).isPresent()) return true;
+
+        return false;
+    }
+
+    private Optional<Inventory> getTeamsInventory(Player bukkitPlayer) {
+        MatchPlayer player = match().getPlayer(bukkitPlayer);
+        Party team = player.getParty();
+
+        if (!team.isParticipating()) return Optional.empty();
+
+        return Optional.of(teamChests.get(team));
+    }
+
+    private Kit getKitForPlayer(MatchPlayer player) {
+        ItemStack stack = new ItemBuilder(item(TOOL_TYPE))
+                .name(ChatColor.DARK_PURPLE + PGMTranslations.t(ITEM_NAME_KEY, player))
+                .lore(ChatColor.DARK_AQUA + PGMTranslations.t(ITEM_LORE_KEY, player))
+                .get();
+
+        ItemKit kit = new SlotItemKit(stack, Slot.Player.forIndex(SLOT_ID));
```

Comment:
You don't need the static variable but add a type comment here for what the `17` means.